planner/core/stats.go

```diff
@@ -85,13 +98,7 @@ func (ds *DataSource) getStatsByFilter(conds expression.CNFExprs) (*property.Sta
 		UsePseudoStats: ds.statisticTable.Pseudo,
 	}
 	for i, col := range ds.Columns {
-		hist, ok := ds.statisticTable.Columns[col.ID]
-		if ok && hist.Count > 0 {
-			factor := float64(ds.statisticTable.Count) / float64(hist.Count)
-			profile.Cardinality[i] = float64(hist.NDV) * factor
-		} else {
-			profile.Cardinality[i] = profile.RowCount * distinctFactor
-		}
+		profile.Cardinality[i] = ds.getColumnNDV(col.ID)
```

Comment:
this seems incorrect, should it be:
```go
hist, ok := ds.statisticTable.Columns[colID]
if ok && hist.Count > 0 {
	profile.Cardinality[i] = float64(hist.Count)
} else {
	profile.Cardinality[i] = ds.statisticTable.Count
}
```
@winoros

Reply:
No, `cardinality` stores NDV not the total row count.

Reply:
it's better to s/Cardinality/NDV/ to eliminate ambiguity